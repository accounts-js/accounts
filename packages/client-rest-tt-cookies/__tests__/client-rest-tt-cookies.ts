import ClientRestTTCookies from '../src';

const defaultClientRestTTCookies = new ClientRestTTCookies();

const denyClientRestTTCookies = new ClientRestTTCookies({
  access: {
    canStore: false
  },
  refresh: {
    canStore: false
  }
});

const accessToken = 'accessTokenTest';
const refreshToken = 'refreshTokenTest';

const tokens = { accessToken, refreshToken };

class MockCookie {
  private str = '';

  get cookie(){
    return this.str;
  }

  set cookie(s){
    this.str += (this.str ? ';' : '') + s;
  }

  public reset(){
    this.str = ''
  }
}


const mockcookie = new MockCookie()


global.document = {};

document.__defineGetter__('cookie', ()=>mockcookie.cookie);
document.__defineSetter__('cookie', (s)=> {mockcookie.cookie = s});


beforeEach(() => {
  mockcookie.reset();
})


describe('ClientRestTTBody', () => {

  describe('setAccessToken', () => {

    it('should set accessToken', () => {
      defaultClientRestTTCookies.setAccessToken({}, {}, accessToken);
      expect(mockcookie.str).toMatchSnapshot()
    })

    it('should not set accessToken if canStore is false', () => {
      denyClientRestTTCookies.setAccessToken({}, {}, accessToken)
      expect(mockcookie.str).toBe('')
    })

  })

  describe('setRefreshToken', () => {

    it('should set refreshToken', () => {
      defaultClientRestTTCookies.setRefreshToken({}, {}, refreshToken)
      expect(mockcookie.str).toMatchSnapshot()
    })

    it('should not set refreshToken if canStore is false', () => {
      denyClientRestTTCookies.setRefreshToken({}, {}, refreshToken)
      expect(mockcookie.str).toBe('')
    })

  })

  describe('setTokens', () => {

    it('should set both Tokens', () => {
      defaultClientRestTTCookies.setTokens({},{},tokens)
      expect(mockcookie.str).toMatchSnapshot()
    })

  })

  describe('getAccessToken', () => {

    it('should get accessToken', () => {
      mockcookie.cookie = 'accessToken=' + accessToken;
      expect(defaultClientRestTTCookies.getAccessToken()).toBe(accessToken);
    })

    it('should return undefined if no accessToken', () => {
      expect(defaultClientRestTTCookies.getAccessToken()).toBe(undefined)
    })

  })

  describe('getRefreshToken', () => {

    it('should get refreshToken', () => {
      mockcookie.cookie = 'refreshToken=' + refreshToken;
      expect(defaultClientRestTTCookies.getRefreshToken()).toBe(refreshToken)
    })

    it('should return undefined if no refreshToken', () => {
      expect(defaultClientRestTTCookies.getRefreshToken()).toBe(undefined)
    })

  })

  describe('getTokens', () => {

    it('should get both Tokens', () => {
      document.cookie = 'accessToken=' + accessToken
      document.cookie = 'refreshToken=' + refreshToken
      expect(defaultClientRestTTCookies.getTokens()).toEqual(tokens)
    })

  })
  
})