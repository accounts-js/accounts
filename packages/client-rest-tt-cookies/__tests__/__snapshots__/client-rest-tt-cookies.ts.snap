// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClientRestTTBody setAccessToken should set accessToken 1`] = `"accessToken=accessTokenTest;max-age=3600000"`;

exports[`ClientRestTTBody setRefreshToken should set refreshToken 1`] = `"refreshToken=refreshTokenTest;max-age=3600000"`;

exports[`ClientRestTTBody setTokens should set both Tokens 1`] = `"accessToken=accessTokenTest;max-age=3600000;refreshToken=refreshTokenTest;max-age=3600000"`;
